### PR TITLE
Move of built-in implementation helpers to jerry-core/ecma/builtin-ob…

### PR DESCRIPTION
**Pull request by bzsolt**
…jects

Related issue: #145

JerryScript-DCO-1.0-Signed-off-by: Zsolt Borbély zsborbely.u-szeged@partner.samsung.com

### CODE REVIEW

File: jerry-core/ecma/builtin-objects/ecma-builtin-array-prototype.cpp [hunk 1]
```diff
@@ -315,6 +314,67 @@ ecma_builtin_array_prototype_object_for_each (ecma_value_t this_arg, /**< this a
...
+  {
+    ecma_string_t *empty_string_p = ecma_get_magic_string (LIT_MAGIC_STRING__EMPTY);
+    ret_value = ecma_make_normal_completion_value (ecma_make_string_value (empty_string_p));
+  }
+  else
+  {
+    ret_value = ecma_op_to_string (index_value);
+  }
+
+  ECMA_FINALIZE (index_value)
```
Comment (ruben-ayrapetyan) [inline jerry-core/ecma/builtin-objects/ecma-builtin-array-prototype.cpp]: Could you, please, add semicolon to the line?

**Comment by ruben-ayrapetyan** [comment]
Looks good to me

**Comment by egavrin** [comment]
Good to me.

**Comment by galpeter** [comment]
lgtm

**Comment by egavrin** [comment]
`make push`

**Comment by dbatyai** [comment]
merged: 69bfcea6975ff07b836945b6ab4a25875daf4530

**Pull request closed without merging**